at_client/lib/src/encryption_service/self_key_encryption.dart

```diff
@@ -9,35 +11,45 @@ import 'package:at_utils/at_logger.dart';
 class SelfKeyEncryption implements AtKeyEncryption {
   final _logger = AtSignLogger('SelfKeyEncryption');
 
+  late LocalSecondary? localSecondary;
+
+  SelfKeyEncryption({this.localSecondary});
+
   @override
   Future<dynamic> encrypt(AtKey atKey, dynamic value) async {
     if (value is! String) {
+      _logger.severe(
+          'Invalid value type found: ${value.runtimeType}. Valid value type is String');
       throw AtEncryptionException(
           'Invalid value type found: ${value.runtimeType}. Valid value type is String');
     }
+    localSecondary ??=
+        AtClientManager.getInstance().atClient.getLocalSecondary();
+    // Get AES key for current atSign
+    var selfEncryptionKey = await _getSelfEncryptionKey(localSecondary!);
+    selfEncryptionKey =
+        DefaultResponseParser().parse(selfEncryptionKey).response;
+    // Encrypt value using sharedKey
+    return EncryptionUtil.encryptValue(value, selfEncryptionKey);
+  }
+
+  Future<String> _getSelfEncryptionKey(LocalSecondary localSecondary) async {
+    String? selfEncryptionKey;
     try {
-      // Get AES key for current atSign
-      var selfEncryptionKey = await _getSelfEncryptionKey();
-      if (selfEncryptionKey == null ||
-          selfEncryptionKey.isEmpty ||
-          selfEncryptionKey == 'data:null') {
-        throw KeyNotFoundException(
-            'Self encryption key is not set for atSign ${atKey.sharedBy}');
+      selfEncryptionKey = await localSecondary.getEncryptionSelfKey();
+      if (selfEncryptionKey.isNull || selfEncryptionKey == 'data:null') {
```

Comment:
local secondary getEncryptionSelfKey has a call to keystore.get..keystore.get will not return data:null.. do we need the check?

Reply:
Yeah, the 'data:null' check can be ignored. I will remove the condition and push the changes.